### DISCUSSION TITLE
How to alias config in `packer.startup{}`

### DISCUSSION THREAD

**Discussion by bangedorrunt** [wbthomason/packer.nvim#443] [Q&A] [answered]
This is my setup which is not working atm
```lua
return require 'packer'.startup {
  function()
    local config = function(name)
      pcall(require, 'plugins.' .. name)
    end
    -- Didn't work!
    use { 'foo', config = config 'foo' }
  end

}
```
FYI, `use  { 'foo', config = function() require 'plugins.foo' end }` worked  as expected

**Comment by wbthomason** [comment, marked as answer]
Right now, your config is passing the result of the `pcall` to the `config` key. You want your config function to return a function that calls `pcall` as desired. However, you might run into trouble with this because `string.dump` does not allow upvalues. #402 will fix that shortcoming, or you can just have your config function return a string instead (which will be inserted verbatim).

**Reply by bangedorrunt** [reply]
Thank you heaps!
